packages/vanilla-renderers/src/complex/categorization/CategorizationRenderer.tsx

```diff
@@ -22,88 +22,105 @@
   OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
   THE SOFTWARE.
 */
-import React from 'react';
+import React, { useMemo, useState } from 'react';
 import type { Categorization, Category, LayoutProps } from '@jsonforms/core';
+import { isVisible } from '@jsonforms/core';
 import {
-  RendererComponent,
   TranslateProps,
+  useJsonForms,
   withJsonFormsLayoutProps,
   withTranslateProps,
 } from '@jsonforms/react';
 import { CategorizationList } from './CategorizationList';
 import { SingleCategory } from './SingleCategory';
-import { isCategorization } from './tester';
 import { withVanillaControlProps } from '../../util';
-import type { VanillaRendererProps } from '../../index';
+import type { AjvProps, VanillaRendererProps } from '../../index';
```

Comment:
We should not import from a parent index

Reply:
I think the issue is, that the types are defined in that index file.
e.g.: https://github.com/eclipsesource/jsonforms/blob/master/packages/vanilla-renderers/src/index.ts#L86-L103

Reply:
That's not good. I would prefer if the `index` files only export and do nothing on their own. Can you refactor this and move the types and renderers each to an own file?

Reply:
Done